packages/slider/src/HandleController.ts

```diff
@@ -196,13 +196,31 @@ export class HandleController implements Controller {
         this.updateModel();
     }
 
+    private waitForUpgrade(handle: HTMLElement): boolean {
+        if (handle instanceof SliderHandle) {
+            return false;
+        }
+        handle.addEventListener(
+            'sp-slider-handle-ready',
+            () => this.extractModelFromLightDom(),
```

Comment:
I think this should be good without batching, but check my logic -

Since `extractModelFromLightDom` bails on the *first* un-upgraded handle, a maximum of one listener will be set up per invocation. Changes to light-DOM children trigger the function, so any subsequent DOM mutations will also set up (at most) one listener. If that listener triggers and then *yet another* handle type hasn't yet upgraded, then *that* handle will set up a single listener and bail, etc, until eventually every handle fulfills `instanceof SliderHandle` and the extraction goes through.

Reply:
> bails on the first un-upgraded handle

This is probably the key to preventing the extra work! Sounds goo to me. Thanks for helping to clarify this, we may want to put a small comment to keep that in mind when coming back to this code.

Reply:
Cool, added!